### PR TITLE
[wip]feat(payments): update keyboard a11y on disabled form

### PR DESCRIPTION
**Pull request by sardesam**
## Because

- Elements are still tabbable while form is disabled.

## This pull request

- Adds `tabindex="-1"` to interactive form elements when disabled and removes attribute to return to DOM sequence for keyboard navigation when enabled.

## Issue that this pull request solves

Closes: FXA-7340

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

**Review by StaberindeZA — CHANGES_REQUESTED** [review]
I see this PR is still in Draft, so I hope I'm not jumping the gun here.

As you know, it's typically not [best practice to access the DOM](https://react.dev/learn/manipulating-the-dom-with-refs#best-practices-for-dom-manipulation-with-refs) directly in React and it's best to avoid.

As an alternative solution, for the React components maintained by SubPlat, is it possible to just set the tabIndex attribute for relevant HTML elements in JSX?

For the elements in the Stripe iframe, I noticed that we aren't actually setting the disabled option of the Stripe Card Element as defined by StripeCardNumberElementOptions. I noticed that once this is set, it looks like the tabIndex's are automatically updated.

**Comment by sardesam** [comment]
@StaberindeZA not jumping the gun at all. I prefer discussion throughout dev :) .

TL;DR: I can make the recommended changes to SubPlat elements but not to Stripe or PayPal.

This was an add-on but not priority (I improved a11y in FXA-7117 but keyboard navigation inconsistencies were still bugging me). Totally aligned. It's not best practice, but it was only intended to be a temporary fix that will be changed in 3.0 when structure and forms are refactored. Updating individual SubPlat elements raises the scope as it goes down all the way to `fields.tsx` (shared in all areas of payments) and `tabindex` does not exist there. Ideally, it's best to leave `tabindex` as is, but in this case, I do need to remove/restore keyboard navigation. I can update Story Points and make the suggested change to SubPlat elements. Stripe and PayPal are a different story.

Adding disabled to Stripe adds the attribute, which is bad practice for a11y because it is invisible to screen readers. Further, you can still tab to the inputs inside the `iframe` in the `StripeElement` container. I could not find a resolution in the official documentation.

PayPal specifically has a `tabindex` set to `0` on a `div` that's embedded in the `iframe` itself. Similar to Stripe, I could not find a resolution in the official documentation.

Inert was another possibility, but it is similar to the `disabled` attribute and invisible to screen readers. :/

Open to other ideas. Would love it if there was a way that doesn't break best practices!

**Comment by StaberindeZA** [comment]
> @StaberindeZA not jumping the gun at all. I prefer discussion throughout dev :) .
> ....

Regarding Stripe, just to make sure we are talking about the same thing, in my initial comment I was referring to the Stripe Card Element `options.disabled`. ([Link to the docs here](https://stripe.com/docs/js/elements_object/create_element?type=card#elements_create-options-disabled)) It seems to do most of the tab index work, except for a scroll area that still seems to be tab-able.

IMO, when we use third party SDKs we should rely on the accessibility options they have available. In the case of Stripe it seems to mostly do what we need.

For PayPal, since its SDK doesn't work as expected, it'd be good to move the accessibility logic closer to the PayPal component. This way, where ever we use the PayPal component, the accessibility logic is included. (This helps with the code duplication in the current approach)

After that, it doesn't look like any other changes are actually required.

**Comment by sardesam** [comment]
Closing as this will be addressed in SP3.
